template-files/action.py

```diff
@@ -53,66 +64,138 @@ class TemplateState(Enum):
     MISSING = "missing"
     USED = "used"
     CONTEXT = "context"
-    WIDTH = nonmember(12)
-
-    def __rich__(self) -> str:
-        if self == self.UNUSED:
-            return f"[yellow]⚠️ ({self.value})[/yellow]"
-        elif self == self.MISSING:
-            return f"[red]❌ ({self.value})[/red]"
-        elif self == self.USED:
-            return f"[green]✅ ({self.value})[/green]"
-        elif self == self.CONTEXT:
-            return f"[blue]📑 ({self.value})[/blue]"
-        else:
-            raise ValueError("Invalid TemplateState")
-
-
-class StubLoader(FileSystemLoader):
-    current: tuple[str, str, str] | None
-    stubs: dict[str, TemplateState]
-    templates: dict[tuple[str, str, str], dict[str, TemplateState]]
-
-    def __init__(self, *args, **kwargs) -> None:
-        super().__init__(*args, **kwargs)
-        self.templates = defaultdict(dict)
-        self.stubs = dict.fromkeys(self.list_templates(), TemplateState.UNUSED)
+    OPTIONAL = "optional"
+
+    @classmethod
+    def from_count(cls, count: int) -> TemplateState:
+        if count < 0:
+            return cls.MISSING
+        if count == 0:
+            return cls.UNUSED
+        return cls.USED
+
+    @classmethod
+    def from_value(cls, value: Any) -> TemplateState:
+        if isinstance(value, Undefined):
+            return cls.MISSING
+        if value is missing:
+            return cls.OPTIONAL
+        return cls.CONTEXT
+
+    @cache
+    def _get_emoji_style(self) -> tuple[str, str]:
+        match self:
+            case self.UNUSED:
+                return ":warning-emoji:", "yellow"
+            case self.MISSING:
+                return ":cross_mark:", "red"
+            case self.USED:
+                return ":white_check_mark:", "green"
+            case self.CONTEXT:
+                return ":books:", "blue"
+            case self.OPTIONAL:
+                return ":heavy_plus_sign:", "yellow"
+            case _:
+                raise ValueError("Invalid TemplateState")
```

Comment:
or is the old-school way more clear? it's more concise 🤷🏼‍♂️

```suggestion
    @cache
    def _get_emoji_style(self) -> tuple[str, str]:
        try:
            return {
                self.UNUSED: (":warning-emoji:", "yellow"),
                self.MISSING: (":cross_mark:", "red"),
                self.USED: (":white_check_mark:", "green"),
                self.CONTEXT: (":books:", "blue"),
                self.OPTIONAL: (":heavy_plus_sign:", "yellow"),
            }[self]
        except KeyError:
            raise ValueError("Invalid TemplateState")
```

Reply:
No, I like the match ... case ... statements.